### PR TITLE
fix/SA-251 - conditional teams and slack resources

### PR DESCRIPTION
**Pull request by wozzer72**
Now only creates slack or teams infrastructure if `slack` or `teams` is defined. The `enable_slack` and `enable_teams` are still used to toggle subscriptions.

Includes removing unused variables as a result of running "tflint" locally.

### CODE REVIEW

File: modules/notify/main.tf [hunk 1]
```diff
@@ -3,7 +3,7 @@ data "aws_partition" "current" {}
 data "aws_region" "current" {}
 
 locals {
-  create = var.create && var.putin_khuylo
+  create = var.create
```
Comment (gambol99) [inline modules/notify/main.tf:6]: haha

File: modules/notify/main.tf [hunk 2]
```diff
@@ -76,10 +76,21 @@ locals {
       warning-icon-url = var.post_icons_url.warning_url
     }
   )
+
+  # the enable_[slack|teams] variable controls the subsubcription between SNS and lambda only; it is
```
Comment (gambol99) [inline modules/notify/main.tf:80]: subsubcription

**Review by gambol99 — APPROVED** [review]
LGTM … minor spelling mistake

File: modules/notify/main.tf (hunk 2 above)
Comment (wozzer72) [reply modules/notify/main.tf]: I've updated and just pushing back.

**Pull request merged**